### PR TITLE
Es/update tags and presets

### PR DESCRIPTION
**Pull request by ErikSin**
Updated presets to align with observation tags.

Most notably changes:
- Creation of the`matchPreset` function. this function matches the tags to a preset returning the best matched preset.
- Removed `preset` property from persisted draft observation. Instead a preset is now matched to the tags in an observation, using the `matchPreset` function. This avoids having 2 sources of truth
- Removed `categoryId` from `observation['tags']`. This was previously used to match the preset with observation, but is no longer needed now that we can use the `matchPreset` function

### CODE REVIEW

File: src/frontend/lib/utils.ts [hunk 1]
```diff
@@ -278,3 +269,47 @@ function leftPad(str: string, len: number, char: string): string {
...
+    for (const key in preset.tags) {
+      if (preset.tags.hasOwnProperty(key)) {
+        const presetTag = preset.tags[key];
+        const availableTag = availableTags[key];
+        if (presetTag === availableTag) {
+          score++;
+          matchedTagsCount++;
+        } else if (
+          Array.isArray(presetTag) &&
+          (presetTag as (boolean | number | string | null)[]).includes(
```
Comment (achou11) [inline src/frontend/lib/utils.ts:294]: Can remove the type casting here:

```suggestion
          presetTag.includes(
```

**Review by achou11 — APPROVED** [review]
most of the comments are minor suggestions for the documentation

File: src/frontend/lib/utils.ts [hunk 2]
```diff
@@ -278,3 +269,47 @@ function leftPad(str: string, len: number, char: string): string {
...
+  let bestMatch: Preset | undefined;
+  let bestMatchScore = 0;
+
+  presets.forEach(preset => {
+    let score = 0;
+    let presetTagsCount = Object.keys(preset.tags).length;
+    let matchedTagsCount = 0;
+
+    for (const key in preset.tags) {
+      if (preset.tags.hasOwnProperty(key)) {
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/lib/utils.ts:286]: I believe (it is really hard to find any documentation) hasOwn [was introduced in ES2022](https://javascript.plainenglish.io/its-time-to-meet-object-hasown-4a26c9de76a). [Hermes explicitly supports es6(2015)](https://hermesengine.dev/docs/language-features/). So hasOwnProperty feels like a safer bet to keep.

File: docs/ObservationPresets.MD
[the remark this answers is not in the document]
Comment (ErikSin) [reply docs/ObservationPresets.MD:1]: https://github.com/digidem/CoMapeo-mobile/pull/186/commits/a332e91376149ae72928193e80d421d06fc4d532

File: src/frontend/lib/utils.ts [hunk 3]
```diff
@@ -278,3 +269,47 @@ function leftPad(str: string, len: number, char: string): string {
 //   // TODO change how we determine whether we are in practice mode or not
 //   return config.metadata.name === "mapeo-default-settings";
 // }
+
+export function matchPreset(
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/lib/utils.ts:280]: https://github.com/digidem/CoMapeo-mobile/pull/186/commits/d9d71968cbb68c4cafd5903bdd67a28fbc9c1773

**Pull request merged**